src/SFA.DAS.EmployerFinance.Types/Models/ExpiredFunds.cs

```diff
@@ -90,16 +90,21 @@ private static void CalculateAndApplyAdjustmentsToTotals(IDictionary<CalendarPer
 
         private static void CalculateAndApplyExpiredFundsToFundsOut(
             IDictionary<CalendarPeriod, decimal> fundsOut, 
-            IDictionary<CalendarPeriod, decimal> expired)
+            IDictionary<CalendarPeriod, decimal> expired,
+            IDictionary<CalendarPeriod, decimal> fundsIn,
+                int expiryPeriod)
         {
-            if (expired == null || !expired.Any(c => c.Value > 0))
+            if (expired == null)
             {
                 return;
             }
            
             foreach (var expiredAmount in expired)
             {
-                var amount = expiredAmount.Value;
+                var levyPeriodForExpiry =
+                    new DateTime(expiredAmount.Key.Year, expiredAmount.Key.Month, 1).AddMonths(expiryPeriod * -1);
+                var amount = fundsIn.FirstOrDefault(c => c.Key.Equals(new CalendarPeriod(levyPeriodForExpiry.Year, levyPeriodForExpiry.Month))).Value - expiredAmount.Value;
```

Comment:
Should this be `Single` instead of `FirstOrDefault`? Or could there be 0 or many funds in instances for the expiry date?

Reply:
Have changed to SingleOrDefault for that one, and Single for the one further in the file